impl/src/test/groovy/com/okta/sdk/impl/ds/cache/DefaultResourceCacheStrategyTest.groovy

```diff
@@ -0,0 +1,177 @@
+        when(cacheRegionNameResolver.getCacheRegionName(StubResource)).thenReturn(StubResource.getName())
+        CacheResolver cacheResolver = new DefaultCacheResolver(cacheManager, cacheRegionNameResolver)
+
+        ResourceCacheStrategy cacheStrategy = new DefaultResourceCacheStrategy(urlResolver, new DefaultResourceHrefResolver(), cacheResolver)
+
+        DefaultCanonicalUri canonicalUri = new DefaultCanonicalUri(resourceUrl, QueryString.create(""))
+        ResourceDataRequest dataRequest = new DefaultResourceDataRequest(ResourceAction.READ, canonicalUri, StubResource, Collections.emptyMap())
+        def payload = createCacheableResourceData(resourceUrl)
+        ResourceDataResult resourceDataResult = new DefaultResourceDataResult(ResourceAction.READ, canonicalUri, StubResource, payload)
+
+        // cache value
+        cacheStrategy.cache(dataRequest, resourceDataResult)
+
+        DefaultCanonicalUri rpcActionUri = new DefaultCanonicalUri("${resourceUrl}/lifecycle/foobar", QueryString.create(""))
+        ResourceDataRequest rpcActionDataRequest = new DefaultResourceDataRequest(ResourceAction.UPDATE, rpcActionUri, canonicalUri, VoidResource, StubResource, Collections.emptyMap(), null)
+        ResourceDataResult rpcActionDataResult = new DefaultResourceDataResult(ResourceAction.UPDATE, rpcActionUri, StubResource, Collections.emptyMap())
+
+        // hit the cache again with the RPC request
+        cacheStrategy.cache(rpcActionDataRequest, rpcActionDataResult)
+
+        verify(cache).remove(resourceUrl)
+    }
+
+    @Test
+    void nonCacheableObjectTest() {
+
+        String baseUrl = "https://okta.example.com/cache-strategy-test"
+        String resourceUrl = "${baseUrl}/cache-me"
+
+        DefaultBaseUrlResolver urlResolver = new DefaultBaseUrlResolver(baseUrl)
+        CacheManager cacheManager = mock(CacheManager)
+        CacheRegionNameResolver cacheRegionNameResolver = mock(CacheRegionNameResolver)
+        when(cacheRegionNameResolver.getCacheRegionName(StubResource)).thenReturn(StubResource.getName())
+        CacheResolver cacheResolver = new DefaultCacheResolver(cacheManager, cacheRegionNameResolver)
+
+        ResourceCacheStrategy cacheStrategy = new DefaultResourceCacheStrategy(urlResolver, new DefaultResourceHrefResolver(), cacheResolver)
+
+        DefaultCanonicalUri canonicalUri = new DefaultCanonicalUri(resourceUrl, QueryString.create(""))
+        ResourceDataRequest dataRequest = new DefaultResourceDataRequest(ResourceAction.READ, canonicalUri, StubResource, Collections.emptyMap())
+        def payload = createNonResourceData()
+        ResourceDataResult resourceDataResult = new DefaultResourceDataResult(ResourceAction.READ, canonicalUri, StubResource, payload)
+
+        // cache value
+        cacheStrategy.cache(dataRequest, resourceDataResult)
+
+        // getCache was never called
+        verifyNoMoreInteractions(cacheManager)
+    }
+
+
```

Comment:
Why do you like the multiple line breaks so much?!

Reply:
It makes me feel like I'm writing a lot of code :P